File: lua src/round_wave.lua
local function sleep(n)
    local runTime = os.clock()

    while os.clock() - runTime <= n do --[[ nothing ]] end
end

local i = 0.0
while true do

   for _=0, 65 * math.abs(math.tan(math.cos(i))) do
      io.write(" ")
   end

   print("*")

   i = i + 0.1

   sleep(0.1)
   
end

File: c src/round_wave.c
#include <stdio.h>
#include <unistd.h>
#include <math.h>
#include <stdlib.h>

int main() {
  double i = 0.0;

  for (;;) {
    for (int h = 0; h < abs(100 * tan(cos(i))); h++) {
      printf(" ");
    }

    printf("*\n");

    i += 0.1;

    usleep(60000);
  }
}
